src/XPath2/XPath2Expression.cs

```diff
@@ -18,20 +18,20 @@ namespace Wmhelp.XPath2
     public class XPath2Expression
     {
         private readonly string expr;
-        private readonly AbstractNode exprTree;
         private readonly XPath2Context context;
         private XPath2ResultType? resultType;
+        public AbstractNode ExpressionTree { get; }

```

Comment:
Maybe also add comment here that this property is only for internal. (Just like the class en delegate comment)

Reply:
Done.